File: apps/src/chat/backend/packets/http/search_user_packet.py
from ..packet import Packet

class SearchUserPacket(Packet):

    def __init__(self, username):
        self._username = username

    def get_type(self):
        return "search_user"

    def to_data(self):
        return {
            "type": self.get_type(),
            "username": self._username
        }

    @staticmethod
    def from_data(data):
        return SearchUserPacket(data["username"])

    def get_username(self) -> str:
        return self._username

File: apps/src/chat/frontend/js/packets/http/SearchUserPacket.js
export default class SearchUserPacket {

    #username;

    constructor(
        username
    ){
        this.#username = username;
    }

    toData(){
        return {
            username: this.#username,
            type: "search_user"
        }
    }

    static fromData(data){
        return new SearchUserPacket(
            data["username"]
        );
    }

    getUsername(){
        return this.#username;
    }
}